_posts/2020-04-21-sparklyr-1.2.0-released/sparklyr-1.2.0.Rmd

```diff
@@ -0,0 +1,130 @@
+---
+title: "sparklyr 1.2: Foreach, Spark 3.0 and Databricks Connect"
+description: |
+  New sparklyr release is now available. This sparklyr 1.2 release features new functionalities such as support for Databricks Connect, a Spark backend for the 'foreach' package, inter-op improvements for working with Spark 3.0 preview, as well as a number of bug fixes and improvements addressing user-visible pain points.
+author:
+  - name: Yitao Li
+    url: https://github.com/yl790
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com
+slug: sparklyr-1.2
+date: 04-21-2020
+categories:
+  - R
+  - Packages/Releases
+  - Distributed Computing
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/sparklyr.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+Behold the glory that is [sparklyr](https://sparklyr.ai) 1.2! In this release, the following new hotnesses have emerged into spotlight:
+
+- A `registerDoSpark` method to create a [foreach]((https://cran.r-project.org/web/packages/foreach/index.html)) parallel backend powered by Spark that enables hundreds of existing R packages to run in Spark.
+- Support for [Databricks Connect](https://docs.databricks.com/dev-tools/databricks-connect.html), allowing `sparklyr` to connect to remote Databricks clusters.
+
+A number of inter-op issues observed with `sparklyr` and Spark 3.0 preview were also addressed recently, in hope that by the time Spark 3.0 officially graces us with its presence, `sparklyr` will be fully ready to work with it. Most notably, key features such as `spark_submit`, `sdf_bind_rows`, and standalone connections are now finally working with Spark 3.0 preview.
+
+To install `sparklyr` 1.2 from CRAN run,
+
+```{r echo=TRUE, eval=FALSE}
+install.packages("sparklyr")
+```
+
+The full list of changes are available in the sparklyr [NEWS](https://github.com/sparklyr/sparklyr/blob/master/NEWS.md) file.
+
+## Foreach
+
+The `foreach` package provides the `%dopar%` operator to iterate over elements in a collection in parallel. Using `sparklyr` 1.2, you can now register Spark as a backend using `registerDoSpark()` and then easily iterate over R objects using Spark:
+
+```{r echo=TRUE, eval=FALSE}
+library(sparklyr)
+library(foreach)
+
+sc <- spark_connect(master = "local")
+
+registerDoSpark(sc)
+foreach(i = 1:3, .combine = 'c') %dopar% {
+  sqrt(i)
+}
+```
+```
+[1] 1.000000 1.414214 1.732051
+```
+
+Now, since many R packages are based on `foreach` to perform parallel computation, we can now make use of all those great packages in Spark as well!
```

Comment:
```suggestion
Since many R packages are based on `foreach` to perform parallel computation, we can now make use of all those great packages in Spark as well!
```